### PR TITLE
please ignore previous pull request, did not properly commit before pushing. this request should be valid now.

### PR DESCRIPTION
**Pull request by Bboymad**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        return request(app).get('/api/topics').expect(200);
+      });
+      test('response contains an array of objects for each topic, which contains the correct properties', () => {
+        return request(app)
+          .get('/api/topics')
+          .expect(200)
+          .then(({ body }) => {
+            expect(body.allTopics).toHaveLength(3);
+            body.allTopics.forEach((topic) => {
+              expect(typeof topic.slug).toBe('string');
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: this is fine but have a look into other jest matchers you can use for asserting against objects in future tickets for example .objectContaining()

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+              expect(typeof topic.description).toBe('string');
+            });
+          });
+      });
+    });
+    describe('errors', () => {
+      test('should respond with 404 when table is empty', () => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: I'm not sure this is an accurate test description, it's checking that an invalid path responds with a 404, nothing to do with the tables

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        return request(app)
+          .get('/api/topicsxxx')
+          .expect(404)
+          .then(({ body }) => {
+            if (body === 0) {
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: you shouldn't be doing any kind of conditional logic inside your tests, did you actually see this test fail? I wouldn't expect body to ever be the value 0 so are you ever getting into this if block? I suspect the test giving you a false positive here

File: app.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+const express = require('express');
+const app = express();
+const { getTopics } = require('./controller/controller')
+
+app.use(express.json());
```
Comment (roisinoreilly) [inline app.js]: we're not using this line yet, please remove it until the code requires you to add it in

File: app.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+// Error Handling
+app.use((err, req, res, next) => {
+    console.error(err);
+  
+    if (err.code) {
+      switch (err.code) {
+        case '23502':
+        case '22P02':
+        case '23503':
```
Comment (roisinoreilly) [inline app.js]: have you come across 23503 yet? I would have thought you would only see this on a future ticket

File: controller/controller.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+const { getAllTopics } = require('../model/model');
+
+exports.getTopics = (req, res, next) => {
+  const params = req.params
```
Comment (roisinoreilly) [inline controller/controller.js]: this variable isn't being used

File: controller/controller.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    getAllTopics(req.params)
+      .then((allTopics) => {
+        console.log(allTopics)
```
Comment (roisinoreilly) [inline controller/controller.js]: remove any logs before pushing

File: model/model.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+const db = require('../db/connection');
+
+exports.getAllTopics = () => {
+  const values = []
```
Comment (roisinoreilly) [inline model/model.js]: is this doing anything at the moment?

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
a good start so far, I'd like you to implement a few changes before I can approve this, most importantly the test that I don't think you saw fail. Don't forget as well that your tests should be driving every line of code you write, you shouldn't be anticipating future tickets or functionality and should be sticking strictly to code that will pass a failing test only. Once you've sorted these few things send us another PR please :)

File: __tests__/app.test.js [hunk 9]
```diff
@@ -18,22 +18,33 @@ describe('GET /api/topics', () => {
...
+                  description: 'what books are made of',
+                  slug: 'paper'
+                }
+              ]))
           });
       });
     });
     describe('errors', () => {
-      test('should respond with 404 when table is empty', () => {
+      test('should respond with 404 when table doesn\'t exist due to invalid endpoint', () => {
```
Comment (Beth-McCormack) [inline __tests__/app.test.js:40]: Please remove mention of tables here as this isnt what the test is asserting, but that the endpoint is invalid

**Review by Beth-McCormack — COMMENTED** [review]
Just left a comment here, feel free to merge after you've had a look!

**Pull request merged**